eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -50,12 +71,37 @@ foreach ($branch in $branches)
     continue
   }
 
-  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request. Deleting Branch"
-  try{
-    Remove-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref ($branch.Remove(0,5)) -AuthToken $AuthToken
-  }
-  catch {
-    LogError "Remove-GitHubSourceReferences failed with exception:`n$_"
-    exit 1
+  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request."
+  # Allow to override the the var $LastCommitOlderThan
+  if ($LastCommitOlderThan -or $env:LASTCOMMITDATEOVERRIDE) {
+    $LastCommitOlderThan = If (!$LastCommitOlderThan) { $env:LASTCOMMITDATEOVERRIDE } Else { $LastCommitOlderThan } 
+    LogDebug "Deleteing the branch which last commit older than the date $LastCommitOlderThan."
+    
+    if (!$res.object -or !$res.object.url) {
+      LogDebug "No commit url returned from response. Skipping... "
+      continue
+    }
+    $GetCommitSHAUrl = $res.object.url
+    $responseJson = HandleGHAPICall -api $GetCommitSHAUrl
+    if ($null -eq $responseJson) {
+      LogError "Error calling the commit SHA API. Exiting..."
+      exit 1
+    }
+    
+    $commitResponse = ConvertFrom-Json $responseJson
+    if (!$commitResponse.author -or !$commitResponse.author.date) {
+      LogDebug "No date returned from the commit sha. Skipping..."
+    }
+    if ($commitResponse.author.date -gt $LastCommitOlderThan) {
+      continue
+    }
+    LogDebug "The branch $branch last commit is older than the date $LastCommitOlderThan. Ready to delete."
   }
+
+  # Remove the branch 
+  $ref = ($branch.Remove(0,5))
```

Comment:
What are we removing here? Is it `refs/` if so we should just replace that so it is clear. Or actually lets not even remove it as we just add it back in the delete command below.